salt/engines/fluent.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+'''
+An engine that reads messages from the salt event bus and pushes
+them onto a fluent endpoint.
+
+.. versionadded:: neon
+
+:Configuration:
+
+All arguments are optional
+
+    Example configuration of default settings
+
+    .. code-block:: yaml
+
+        engines:
+          - fluent:
+              host: localhost
+              port: 24224
+              app: engine
+
+    Example fluentd configuration
+
+    .. code-block:: none
+
+        <source>
+            @type forward
+            port 24224
+        </source>
+
+        <match saltstack.**>
+            @type file
+            path /var/log/td-agent/saltstack
+        </match>
+
+:depends: fluent-logger
+'''
+
+# Import python libraries
+from __future__ import absolute_import, print_function, unicode_literals
+import logging
+
+# Import salt libs
+import salt.utils.event
+
+# Import third-party libs
+try:
+    from fluent import sender, event
+except ImportError:
+    handler = None
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'fluent'
+
+
+def __virtual__():
+    return __virtualname__ \
+        if sender is not None \
+        else (False, 'fluent-logger not installed')
+
+
+def start(host='localhost', port=24224, app='engine'):
+    '''
+    Listen to salt events and forward them to fluent
+    '''
```

Comment:
Please add some docs here about the parameters using the Google Style Python Docstrings: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
Done!